### ISSUE TITLE
Travis CI currently fails

### ISSUE THREAD

**Issue by thomasjfox** [file/file-tests#3]
Testing with Travis CI currently fails.
tl;dr: It picks up the wrong libmagic.so:
`file: compiled magic version [532] does not match with shared library magic version [514]`

.travis.yml basically does a "make" and "make install".
After that it calls "file --version" and from the log you can see it's the current file version.
It does not set any LD_LIBRARY_PATH. What happens now is that it will pick up the "libmagic.so" from Travis CI's system instead of the just built one.

We could either tweak around the LD_LIBRARY_PATH *or* set -rpath during compile time.
Something like "LDFLAGS=-Wl,-rpath $BUILD_DIR". That's how I do it for my local file-tests invocation.

Example log output showing the problem:
https://travis-ci.org/file/file-tests/jobs/272927961

$ file --version
file-5.32
magic file from /etc/magic:/usr/share/misc/magic

The command "file --version" exited with 0.

python update-db.py -m file/magic/Magdir
using file from /usr/bin/file
version is file-5.32
magic file from /etc/magic:/usr/share/misc/magic
[ ################################################ ] 100% Splitting patterns 
file: compiled magic version [532] does not match with shared library magic version [514]
file: Unknown !: entry `!:ext cwk'
.. [CRASH] ..

**Comment by glensc** [comment]
the `make install` was made there with the purpose of replacing system version from installed one. an easy solution quickly made.

i see two solutions to this:
1. cleanup system leftovers of `file` package install
2. be independent of `file` package present in the os

"2" is better solution as it would make the test suite usable outside travis too (developers machines), "1" is probably just easier to implement (`apt-get purge file` perhaps).

**Comment by thomasjfox** [comment]
On Friday, 08 September 2017 12:26:30 CEST Elan Ruusamäe wrote:
> the `make install` was made there with the purpose of replacing system
> version from installed one. an easy solution quickly made.
> 
> i see two solutions to this:
> 1. cleanup system leftovers of `file` package install
> 2. be independent of `file` package present in the os
> 
> "2" is better solution as it would make the test suite usable outside travis
> too (developers machines), "1" is probably just easier to implement
> (`apt-get purge file` perhaps).

probably dependencies of other packages will get in the way of 1).

Let's do the second solution, either using -rpath
or by setting the LD_LIBRARY_PATH to the build directory.

The cmdline help of fast-regression-test.py even shows
how to use an alternative location for the magic *database*.

**Comment by glensc** [comment]
ok, i made the 1" solution so far, seems working:
https://travis-ci.org/file/file-tests/builds/273411358

can't test osx part because travis macos infrastructure is down:
https://www.traviscistatus.com/
https://www.traviscistatus.com/incidents/4qylrqvy50gy

this doesn't mean the "2" can't be made as well :)

**Comment by zoulasc** [comment]
this has been fixed a long time ago.
